### ISSUE TITLE
High CPU RAM usage while loading dataset, not dumped after training

### ISSUE THREAD

**Issue by aritejhg** [ultralytics/ultralytics#2530]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

I have noticed that when training for object detection, the RAM usage goes up during the cache creation and first epoch (approximately, Colab doesn't give the most granular reporting) and after that the RAM usage only increases minimally. This in itself is fine, but the problem is that RAM is not cleared after training. This is especially a problem as I am trying to do 10-fold validation, and wanted to do it in 1 single for-loop instead of multiple loops. I'll try to come up with a MRE, meanwhile any suggestions are appreciated :) 
@AyushExel @glenn-jocher 

### Environment

Colab, with the latest ultralytics. 

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@aritejhg oh this is interesting. Are you running any special cache arguments like `cache ram` or `cache disk`?

**Comment by aritejhg** [comment]
hey @glenn-jocher, nope, the only argument passed was epochs and imgsz. I have put my code below as reference. 

`from sklearn.model_selection import KFold

def train_model(train_imgs, val_imgs, fold_number, epochs=100, imgsz=640):
    model = YOLO('yolov8n.pt')
 
    # create folder for fold and copy appropriate files
    os.makedirs(f'./datasets/fold{fold_number}/train/images', exist_ok = True)
    os.makedirs(f'./datasets/fold{fold_number}/train/labels', exist_ok = True)
    os.makedirs(f'./datasets/fold{fold_number}/valid/images', exist_ok = True)
    os.makedirs(f'./datasets/fold{fold_number}/valid/labels', exist_ok = True)
    for file in train_imgs:
      shutil.copy(file, f'./datasets/fold{fold_number}/train/images')
      shutil.copy(file.replace("images", "labels").replace('.jpg','.txt'),f'./datasets/fold{fold_number}/train/labels')
    for file in val_imgs:
      shutil.copy(file, f'./datasets/fold{fold_number}/valid/images')
      shutil.copy(file.replace("images", "labels").replace('.jpg','.txt'),f'./datasets/fold{fold_number}/valid/labels')

    shutil.copy('./datasets/k_fold_dataset/data.yaml', f'./datasets/fold{fold_number}/')
    # Train the model
    model.train(data=f'/content/datasets/fold{fold_number}/data.yaml', epochs=epochs, imgsz=imgsz, stream = True)
    
    return model

def k_fold_cross_validation(k, X, epochs, imgsz):
    kf = KFold(n_splits=k, shuffle=True, random_state=4)
    fold_number = 1
    
    for train_idx, val_idx in kf.split(X):
        print(f"Training on fold {fold_number}")
        print(len(train_idx), len(val_idx))
        train_imgs = [f'/content/datasets/k_fold_dataset/images/{X[i]}' for i in train_idx]
        val_imgs = [f'/content/datasets/k_fold_dataset/images/{X[i]}' for i in val_idx]
        model = train_model(train_imgs, val_imgs, fold_number, epochs, imgsz)
        # model.save(f'yolov8n_fold{fold_number}.pt')
        
        print(f'------------------------------FOLD {fold_number} DONE-------------------------------')

        fold_number += 1


k = 10
X = os.listdir('./datasets/k_fold_dataset/images')
imgsz = 640
epochs = 2
k_fold_cross_validation(k, X, epochs, imgsz)`

**Comment by glenn-jocher** [comment]
@aritejhg, it looks like the cache is not being cleared after model training. One way to clear the cache would be to destroy the pytorch session using `torch.cuda.empty_cache()`. You can call this method after each fold has completed training. Let me know if you have any other issues.

**Comment by aritejhg** [comment]
@glenn-jocher the problem is with CPU RAM usage though, this helps only with the GPU RAM usage.

**Comment by glenn-jocher** [comment]
@aritejhg, my apologies for the confusion regarding RAM usage. Thank you for pointing that out. One way to reduce RAM usage is by decreasing the image size (imgsz) or the batch size during training. Another way is to use `del` to delete objects that are no longer needed in memory, for example, data loaders, and datasets. 

Regarding your side note, you can pass the exact file names for train and test by modifying the `train` and `val` keys in the YAML file to include the desired image file paths. You can refer to the [YOLOv8 docs](https://docs.ultralytics.com/yolov8/modes/) for more information on the YAML file syntax. Let me know if this resolves your issue.
